### PR TITLE
Set request object for FluidEmail

### PR DESCRIPTION
**Pull request by linawolf**
Depends on https://github.com/TYPO3-Documentation/render-guides/pull/455

releases: main, 12.4

### CODE REVIEW

File: Documentation/ApiOverview/Mail/Index.rst [hunk 1]
```diff
@@ -374,6 +374,33 @@ In Fluid, you can now use the defined language key ("language"):
...
+    use TYPO3\CMS\Core\Mail\FluidEmail;
+
+    $email = new FluidEmail();
+    $email->setRequest($this->request);
+
+Read more aboout :ref:`Getting the PSR-7 request object <getting-typo3-request-object>` in different 
+contexts. In a context where no valid request object can be retrieved, such as in a 
+:ref:`Console command <t3coreapi:symfony-console-commands>` the affected ViewHelpers cannot be used.
+
+Trying to use these ViewHelpers without a valid requst throws an :ref:`error <t3exceptions:Exceptions/1639819269>`
```
Comment (brotkrueml) [inline Documentation/ApiOverview/Mail/Index.rst:397]:
```suggestion
Trying to use these ViewHelpers without a valid request throws an :ref:`error <t3exceptions:Exceptions/1639819269>` 
```

File: Documentation/ApiOverview/Mail/Index.rst [hunk 2]
```diff
@@ -374,6 +374,33 @@ In Fluid, you can now use the defined language key ("language"):
...
+    $email->setRequest($this->request);
+
+Read more aboout :ref:`Getting the PSR-7 request object <getting-typo3-request-object>` in different 
+contexts. In a context where no valid request object can be retrieved, such as in a 
+:ref:`Console command <t3coreapi:symfony-console-commands>` the affected ViewHelpers cannot be used.
+
+Trying to use these ViewHelpers without a valid requst throws an :ref:`error <t3exceptions:Exceptions/1639819269>` 
+like the following:
+
+..  code-block:: plain
```
Comment (brotkrueml) [inline Documentation/ApiOverview/Mail/Index.rst:400]:
```suggestion
..  code-block:: text
```

Either "plaintext" or "text"

File: Documentation/ApiOverview/Mail/Index.rst [hunk 3]
```diff
@@ -374,6 +374,33 @@ In Fluid, you can now use the defined language key ("language"):
...
+..  code-block:: php
+
+    use TYPO3\CMS\Core\Mail\FluidEmail;
+
+    $email = new FluidEmail();
+    $email->setRequest($this->request);
+
+Read more aboout :ref:`Getting the PSR-7 request object <getting-typo3-request-object>` in different 
+contexts. In a context where no valid request object can be retrieved, such as in a 
+:ref:`Console command <t3coreapi:symfony-console-commands>` the affected ViewHelpers cannot be used.
```
Comment (brotkrueml) [inline Documentation/ApiOverview/Mail/Index.rst:395]: Are you sure? You can construct a request object manually. What about the password forgot command? Does it not use FluidEmail?

Comment (linawolf) [reply Documentation/ApiOverview/Mail/Index.rst:395]: You can send FluidEmails without request object but not use certain viewhelpers then.

**Review by brotkrueml — APPROVED** [review]

**Comment by brotkrueml** [comment]
@linawolf Please have a look at the tests, as they fail.

**Review by brotkrueml — APPROVED** [review]

**Pull request merged**